src/app/components/multiselect/multiselect.ts

```diff
@@ -834,18 +834,17 @@ export class MultiSelect implements OnInit,AfterViewInit,AfterContentInit,AfterV
 
     updateLabel() {
         if (this.value && this.options && this.value.length && this.displaySelectedLabel) {
-            let label = '';
-            for (let i = 0; i < this.value.length; i++) {
-                let itemLabel = this.findLabelByValue(this.value[i]);
-                if (itemLabel) {
-                    if (label.length > 0) {
-                        label = label + ', ';
+            if (this.value.length <= this.maxSelectedLabels) {
+                let label = '';
+                for (let i = 0; i < this.value.length; i++) {
+                    let itemLabel = this.findLabelByValue(this.value[i]);
```

Comment:
@billyan2018
Thanks for your changes, I think we can determine the range of labels to use at the time of splicing the strings, which will ensure higher efficiency. The range should be the smallest of `maxSelectedLabels` and `value.length`.
```suggestion
const labelScope = Math.min(this.value.length, this.maxSelectedLabels);
for (let i = 0; i < labelScope; i++) {
      let itemLabel = this.findLabelByValue(this.value[i]);
```